### PR TITLE
Use TranslationInterface from Contracts

### PR DESCRIPTION
**Pull request by franmomu**
##  Subject

Removing dependencies with deprecated classes.

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Deprecated
- Constructing `SonataAdminExtension` with an instance of TranslationInterface from components
```

### CODE REVIEW

File: src/Twig/Extension/SonataAdminExtension.php [hunk 1]
```diff
@@ -83,17 +84,41 @@ class SonataAdminExtension extends AbstractExtension
...
+                    sprintf(
+                        'The $translator parameter should be an instance of "%s" and will be mandatory in 4.0.',
+                        TranslatorInterface::class
+                    ),
+                    E_USER_DEPRECATED
+                );
+            }
+
+            if (!$translator instanceof TranslatorInterface && !$translator instanceof LegacyTranslationInterface) {
+                throw new \InvalidArgumentException(
```
Comment (core23) [inline src/Twig/Extension/SonataAdminExtension.php]: Maybe use `TypeError` here?

File: src/Twig/Extension/SonataAdminExtension.php [hunk 2]
```diff
@@ -83,17 +84,41 @@ class SonataAdminExtension extends AbstractExtension
...
             );
+        } else {
+            if (!$translator instanceof TranslatorInterface) {
+                @trigger_error(
+                    sprintf(
+                        'The $translator parameter should be an instance of "%s" and will be mandatory in 4.0.',
+                        TranslatorInterface::class
+                    ),
+                    E_USER_DEPRECATED
+                );
```
Comment (greg0ire) [inline src/Twig/Extension/SonataAdminExtension.php]:
```suggestion
                @trigger_error(sprintf(
                    'The $translator parameter should be an instance of "%s" and will be mandatory in 4.0.',
                    TranslatorInterface::class
                ), E_USER_DEPRECATED);
```

File: src/Twig/Extension/SonataAdminExtension.php [hunk 3]
```diff
@@ -83,17 +84,41 @@ class SonataAdminExtension extends AbstractExtension
...
+
+            if (!$translator instanceof TranslatorInterface && !$translator instanceof LegacyTranslationInterface) {
+                throw new \TypeError(
+                    sprintf(
+                        'Argument 2 must be an instance of "%s" or "%s", "%s given"',
+                        LegacyTranslationInterface::class,
+                        TranslatorInterface::class,
+                        \get_class($translator)
+                    )
+                );
```
Comment (greg0ire) [inline src/Twig/Extension/SonataAdminExtension.php]:
```suggestion
                throw new \TypeError(sprintf(
                    'Argument 2 must be an instance of "%s" or preferably "%s", "%s given"',
                    TranslatorInterface::class,
                    LegacyTranslationInterface::class,
                    \get_class($translator)
                ));
```

File: tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 4]
```diff
@@ -267,6 +268,35 @@ public function setUp(): void
...
+    /**
+     * @doesNotPerformAssertions
+     * @group legacy
+     */
+    public function testConstructWithLegacyTranslator(): void
+    {
+        new SonataAdminExtension(
+            $this->pool,
+            null,
+            $this->createMock(LegacyTranslatorInterface::class)
```
Comment (greg0ire) [inline tests/Twig/Extension/SonataAdminExtensionTest.php]:
```suggestion
            $this->createStub(LegacyTranslatorInterface::class)
```

**Comment by greg0ire** [comment]
Thanks @franmomu !

**Pull request merged**